src/neo/Wallets/NEP6/NEP6Account.cs

```diff
@@ -83,5 +85,52 @@ public bool VerifyPassword(string password)
                 return false;
             }
         }
+
+        /// <summary>
+        /// Cache draft nep2key during wallet password changing process. Should not be called alone for a single account
+        /// </summary>
+        internal bool ChangePasswordPrelude(string password_old, string password_new)
+        {
+            if (WatchOnly) return true;
+            if (nep2key == null)
+            {
+                if (key == null)
+                {
+                    return true;
+                }
+            }
+            else
+            {
+                try
+                {
+                    key = new KeyPair(Wallet.GetPrivateKeyFromNEP2(nep2key, password_old, wallet.Scrypt.N, wallet.Scrypt.R, wallet.Scrypt.P));
+                }
+                catch
+                {
+                    return false;
+                }
+            }
+            nep2keyDraft = key.Export(password_new, wallet.Scrypt.N, wallet.Scrypt.R, wallet.Scrypt.P);
+            isChangingPassword = true;
```

Comment:
we can use `nep2keyDraft != null` as flag, instead of `isChangingPassword`

Reply:
Agree. And `nep2keyDraft` can be renamed to `nep2KeyNew`.

Reply:
Have modified correspondingly, please have a look:)